runtime/lua/warn.lua

```diff
@@ -0,0 +1,25 @@
+-- off by default
+warn("discarded warning")
+
+warn("@on")
+
+warn("a warning")
+--> =Test warning: a warning
+
+warn("this", 12, 9.5)
+--> =Test warning: this129.5
+
+warn("some", "@on", "other", "@off")
+--> =Test warning: some@onother@off
+
+warn("@off")
+
+warn("discarded")
+print("hello")
+--> =hello
+
+print(pcall(warn))
+--> ~false\t.*value needed
+
+print(pcall(warn, {}))
+--> ~false\t.*string expected
```

Comment:
We should probably avoid keeping duplicate tests (see `lib/base/lua/warn.lua`).

Reply:
Yeah I agree, but in this case it doesn't matter if the tests diverge in the future (strictly speaking they are not testing the same code!)

Reply:
Ah ok, thanks for the clarification, clearly I need to be more vigilant about what's going on in the testing :)

Reply:
No worries!